lightning/src/chain/channelmonitor.rs

```diff
@@ -1459,22 +1479,77 @@ impl<Signer: Sign> ChannelMonitor<Signer> {
+							us.onchain_tx_handler.is_output_spend_pending(&
+								if let Some(txid) = htlc_spend_txid_opt {
+									debug_assert!(
+										us.onchain_tx_handler.channel_transaction_parameters.opt_anchors.is_none(),
+										"This code needs updating for anchors");
+									BitcoinOutPoint::new(txid, 0)
+								} else {
+									BitcoinOutPoint::new(confirmed_txid.unwrap(), htlc_commitment_tx_output_idx)
+								});
+
 						if let Some(conf_thresh) = delayed_output_pending {
 							debug_assert!($holder_commitment);
 							res.push(Balance::ClaimableAwaitingConfirmations {
 								claimable_amount_satoshis: htlc.amount_msat / 1000,
 								confirmation_height: conf_thresh,
 							});
-						} else if htlc_resolved.is_some() {
+						} else if htlc_resolved.is_some() && !htlc_output_needs_spending {
 							// Funding transaction spends should be fully confirmed by the time any
 							// HTLC transactions are resolved, unless we're talking about a holder
 							// commitment tx, whose resolution is delayed until the CSV timeout is
 							// reached, even though HTLCs may be resolved after only
 							// ANTI_REORG_DELAY confirmations.
 							debug_assert!($holder_commitment || us.funding_spend_confirmed.is_some());
+						} else if $counterparty_revoked_commitment {
+							let htlc_output_claim_pending = us.onchain_events_awaiting_threshold_conf.iter().find_map(|event| {
+								if let OnchainEvent::MaturingOutput {
+									descriptor: SpendableOutputDescriptor::StaticOutput { .. }
+								} = &event.event {
+									if event.transaction.as_ref().map(|tx| tx.input.iter().any(|inp| {
+										if let Some(htlc_spend_txid) = htlc_spend_txid_opt {
+											Some(tx.txid()) == htlc_spend_txid_opt ||
+												inp.previous_output.txid == htlc_spend_txid
+										} else {
+											Some(inp.previous_output.txid) == confirmed_txid &&
+												inp.previous_output.vout == htlc_commitment_tx_output_idx
+										}
+									})).unwrap_or(false) {
+										Some(())
+									} else { None }
+								} else { None }
+							});
+							if htlc_output_claim_pending.is_some() {
+								// We already push `Balance`s onto the `res` list for every
+								// `StaticOutput` in a `MaturingOutput` in the revoked
+								// counterparty commitment transaction case generally, so don't
+								// need to do so again here.
+							} else if htlc_spend_pending.is_some() {
+								res.push(Balance::CounterpartyRevokedOutputClaimable {
+									claimable_amount_satoshis: htlc.amount_msat / 1000,
+								});
+							} else if htlc_resolved.is_some() {
+								res.push(Balance::CounterpartyRevokedOutputClaimable {
+									claimable_amount_satoshis: htlc.amount_msat / 1000,
+								});
+							} else {
+								debug_assert!(htlc_update_pending.is_none(),
+									"HTLCUpdate OnchainEvents should never appear for preimage claims");
+								debug_assert!(htlc_spend_pending.is_none() || !htlc_spend_pending.unwrap().1,
```

Comment:
Hm, this `debug_assert` will always be true since we checked `htlc_spend_pending` above